File: benchmarks/benchmarks/02_numerical_computation/comparisons/polynomial_eval.py
def polynomial_cost(points: int, degree: int) -> int:
    checksum = 0
    for point in range(points):
        x = (point % 97) + 3
        value = 1
        for coeff in range(degree, 0, -1):
            value = ((value * x) + ((coeff * 11) + (point % 29))) % 1_000_003
        checksum += value
    return checksum


if __name__ == "__main__":
    print(polynomial_cost(800000, 16))

File: benchmarks/benchmarks/02_numerical_computation/comparisons/polynomial_eval.go
package main

import "fmt"

func polynomialCost(points int64, degree int64) int64 {
	var checksum int64
	for point := int64(0); point < points; point++ {
		x := (point % 97) + 3
		value := int64(1)
		for coeff := degree; coeff > 0; coeff-- {
			value = ((value * x) + ((coeff * 11) + (point % 29))) % 1000003
		}
		checksum += value
	}
	return checksum
}

func main() {
	fmt.Println(polynomialCost(800000, 16))
}
